sqlmesh/core/node.py

```diff
@@ -139,10 +154,28 @@ def _cron_validator(cls, v: t.Any) -> t.Optional[str]:
                 raise ConfigError(f"Invalid cron expression '{cron}'")
         return cron
 
-    @validator("owner", "description", "stamp", "interval_unit_", pre=True)
+    @validator("owner", "description", "stamp", pre=True)
     def _string_validator(cls, v: t.Any) -> t.Optional[str]:
         return str_or_exp_to_str(v)
 
+    @validator("interval_unit_", pre=True)
+    def _interval_unit_validator(
+        cls, v: t.Any, values: t.Dict[str, t.Any]
+    ) -> t.Optional[IntervalUnit]:
+        if v is None:
+            return v
+
+        if not isinstance(v, IntervalUnit):
+            v = IntervalUnit(t.cast(str, str_or_exp_to_str(v)).lower())
+        cron = values.get("cron")
```

Comment:
I think `cron` won't be there. I think `values` only contains values that haven't been validated yet and since `interval_unit_` goes after `cron`, it won't be there. I suggest to perform this validation in the `_root_validator`

Reply:
Even if this works right now I don't think it would work with Pydantic v2.

Reply:
I set a breakpoint in tests and verified cron was there, which surprised me too, so I kept it to the validator. I'm fine putting in a root validator instead.